source/adios2/CMakeLists.txt

```diff
@@ -104,7 +104,7 @@ target_sources(adios2 PRIVATE
     engine/dataman/DataManWriter.h
     engine/dataman/DataManWriter.tcc
 )
-target_link_libraries(adios2 PRIVATE NLohmannJson)
+target_link_libraries(adios2 PRIVATE nlohmann_json)
```

Comment:
For some reason, the `nlohmann_json` target is not defined here when build with external

Reply:
Found the reason: `add_subdirectory` keeps found CMake Config/Module packages in the current dir or below... it's not intended to move it to parent (use `include` instead):
https://stackoverflow.com/a/48741931/2719194

Reply:
This is a general bug with `ADIOS2_USE_SYSTEM` packages - will add a fix

Reply:
Issue identified, fix commited: [SO link](https://stackoverflow.com/a/48741931/2719194)

Reply:
(side note: SYSTEM is the wrong naming, it should be EXTERN(AL) or something :-) - in most cases it's hopefully a package manager not the system path ;-) )

Reply:
I agree that the `PackageName_USE_SYSTEM_DependencyName' doesn't really convey what we want here but it's a fairly established convention in the CMake ecosystem for how to specify this concept.